projects/igniteui-angular/src/lib/grids/filtering/grid-filtering.service.ts

```diff
@@ -114,8 +114,14 @@ export class IgxFilteringService implements OnDestroy {
                 }
             });
 
+        this._overlayService.onClosing.pipe(
+            first((overlay) => overlay.id === this._componentOverlayId),
+            takeUntil(this.destroy$)).subscribe(() => {
+                overlayOpening.unsubscribe();
+            });
+
```

Comment:
Do we need to unsubscribe the `overlayOpening`?

Reply:
It's seems that we do not need that, so removing it.